Rakefile

```diff
@@ -262,6 +263,399 @@ task :xcode => [:dependencies] do
+        sh "brew install #{tool}"
+      end
+
+    #End namespace addons
+    end
+
+  #End namespace Tools
+  end
+
+  namespace :lint do
+    task :check do
+      if !git_initialized?
+        puts "Initializing git repository"
+        sh "git init", verbose: false
+      end
+
+      Rake::Task["git:install_hooks"].invoke
+    end
+
+    def git_initialized?
+      sh "git rev-parse --is-inside-work-tree > /dev/null 2>&1", verbose: false
+    end
+  end
+
+#End namespace install
+end
+
+#Credentials deals with the setting up the developer's WPCOM API app ID and app Secret
+namespace :credentials do
+  task :setup => %w[credentials:prompt credentials:set_app_secrets]
+
+  task :prompt do
+    puts ""
+    puts "====================================================================================="
+    puts "To be able to log into the WordPress app while developing you will need to setup API credentials"
+    puts "To do this follow these steps"
+    puts ""
+    puts ""
+    puts ""
+    puts "====================================================================================="
+
+    puts "1. Create a WordPress.com account at https://wordpress.com/start/user (if you don't already have one)."
+    prompt_for_continue
+
+    puts "====================================================================================="
+    puts "2. Create an application at https://developer.wordpress.com/apps/."
+    prompt_for_continue
+
+    puts "====================================================================================="
+    puts '3. Set "Redirect URLs"= https://localhost and "Type" = Native and click "Create" then "Update".'
+    prompt_for_continue
+
+    puts "====================================================================================="
+    puts "4. Copy the Client ID and Client Secret from the OAuth Information."
+    prompt_for_continue
+  end
+
+  def prompt_for_continue
+    STDOUT.puts "Please press enter to continue"
+    STDIN.gets.strip
+  end
+
+  #user given app id and secret and create a new wpcom_app_credentials file
+  task :set_app_secrets do
+    create_secrets_file
+    set_app_secrets(get_client_id, get_client_secret)
+    remove_temp_file
+  end
+
+  def get_client_id
+    STDOUT.puts "Please enter your Cliet id"
+    STDIN.gets.strip
+  end
+
+  def get_client_secret
+    STDOUT.puts "Please enter your Client Secret"
+    STDIN.gets.strip
+  end
+
+  #create temporary secrets file from example file
+  def create_secrets_file
+    sh "cp WordPress/Credentials/wpcom_app_credentials-example .configure-files/temp_wpcom_app_credentials", verbose: false
+  end
+
+  #create a new wpcom_app_credentials file combining the app secret and app id
+  def set_app_secrets(id, secret)
+    puts "Creating credentials file"
+    new_file = File.new(".configure-files/wpcom_app_credentials", "w")
+    File.open(".configure-files/temp_wpcom_app_credentials") do |file|
+      file.each_line do |line|
+        string = line.to_s()
```

Comment:
Is this conversion to String necessary?

Reply:
huh.... could have sworn that I was getting errors before when the to_s wasn't there... but removed it and it is working now...  Simplified that section.  Thanks!!